src/sage/libs/coxeter3/coxeter.pyx

```diff
@@ -86,7 +86,7 @@ cdef class String:
             sage: all([tb > ta1, tb >= ta1, tb >= tb])                 # optional - coxeter3
             True
         """
-        if type(other) != type(self):
+        if type(other) is not type(self):
```

Comment:
Not the fault of this change, but this should probably do `return op == Py_NE` in this case (and possibly return `NotImplemented` for LT, LE, GT, GE)

Reply:
How about
```
        if type(other) is not type(self):
            if op in (Py_LT, Py_LE, Py_GT, Py_GE):
                return NotImplemented
            return op == Py_NE
```
?

Reply:
Yes, LGTM